controllers/gateway_controller.go

```diff
@@ -99,37 +96,63 @@ func (r *GatewayReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		return ctrl.Result{}, nil
 	}
 
+	gatewayReadyStatus, gatewayReadyStatusIsSet := isGatewayReady(gw)
+	oldGateway := gw.DeepCopy()
+	initGatewayStatus(gw)
+	factorizeStatus(gw, oldGateway)
```

Comment:
I'm not seeing a need to do this, especially if you take the suggestion above, can we just drop these? Let me know if I'm missing something :thinking: 

```suggestion
```

Reply:
they are needed to initialize the status and to avoid useless transitions (patch is effective for conditions which differ in the `transitionTIme`, and we don't want this behavior)